### DISCUSSION TITLE
Publishing to the dashboard to Rconnect Server

### DISCUSSION THREAD

**Discussion by sutsabs** [OHDSI/DataQualityDashboard#494] [Q&A]
Is the DQ dashboard shiny app publishable to rconnect server? We want to share the dashboard with the team without having to share the package and the results file. 

I tried to dig into the shiny app file but was not able to successfully publish.

`── Preparing for deployment ────────────────────────────────────────────────────
✔ Deploying "dataquality" to "server: usvaprdrc014.astellasrwi.us / username: A4034337"
ℹ Creating application on server...
✔ Created application with id 90
ℹ Bundling 99 files: 'app.R', 'www/css/resume.css', 'www/favicon.ico', 'www/htmlwidgets/dqDashPOC.js', 'www/htmlwidgets/dqDashPOC.yaml', 'www/htmlwidgets/lib/dqCheckRenderer.js', 'www/htmlwidgets/lib/dqDashboardComponent.js', 'www/htmlwidgets/lib/dqDashboardTemplate.hbs', 'www/htmlwidgets/lib/dqDashboardTemplate.hbs.json', 'www/htmlwidgets/lib/dqMetadataComponent.js', 'www/htmlwidgets/lib/dqMetadataHeading.js', 'www/htmlwidgets/lib/handlebars-latest.js', 'www/img/close.png', 'www/img/details_close.png', 'www/img/details_open.png', 'www/img/ohdsi.png', 'www/img/open.png', 'www/index.html', …, 'www/vendor/jquery-easing/jquery.easing.js', and 'www/vendor/jquery-easing/jquery.easing.min.js'
ℹ Capturing R dependencies with renv
The following package(s) were installed from an unknown source:
- DataQualityDashboard [2.4.0]
renv may be unable to restore these packages in the future.
Consider reinstalling these packages from a known source (e.g. CRAN).

Traceback (most recent calls last):
9: rsconnect::deployApp(appDir = "/opt/R/4.2.2/lib/R/library/DataQualityDashboard/shinyApps", 
       appFileManifest = "/tmp/b9ae-96b4-95ed-22c9", account = "A4034337", 
       server = "usvaprdrc014.astellasrwi.us", appName = "dataquality", 
       appTitle = "DataQuality", launch.browser = function(url) {
           message("Deployment completed: ", url)
       }, lint = FALSE, metadata = list(asMultiple = FALSE, asStatic = FALSE))
8: bundleApp(appName = target$appName, appDir = appDir, appFiles = appFiles, 
       appMetadata = appMetadata, quiet = quiet, verbose = verbose, 
       pythonConfig = pythonConfig, image = image, envManagement = envManagement, 
       envManagementR = envManagementR, envManagementPy = envManagementPy)
7: createAppManifest(appDir = bundleDir, appMetadata = appMetadata, 
       users = users, pythonConfig = pythonConfig, retainPackratDirectory = TRUE, 
       image = image, envManagement = envManagement, envManagementR = envManagementR, 
       envManagementPy = envManagementPy, verbose = verbose, quiet = quiet)
6: bundlePackages(bundleDir = appDir, extraPackages = extraPackages, 
       verbose = verbose, quiet = quiet)
5: computePackageDependencies(bundleDir, extraPackages, quiet = quiet, 
       verbose = verbose)
4: snapshotRenvDependencies(bundleDir, extraPackages, verbose = verbose)
3: renv::snapshot(bundleDir, packages = deps$Package, prompt = FALSE)
2: renv_snapshot_validate_report(valid, prompt, force)
1: stop("aborting snapshot due to pre-flight validation failure")
Error in renv_snapshot_validate_report(valid, prompt, force) : 
  aborting snapshot due to pre-flight validation failure
Calls: <Anonymous> ... snapshotRenvDependencies -> <Anonymous> -> renv_snapshot_validate_report
Execution halted`

**Comment by katy-sadowski** [comment]
Hi, unfortunately I don't have experience deploying apps (including DQD) to Connect; however if you post in the Forums others might have advice.

However - looking at your error, it seems like this may have actually gotten raised when trying to take an renv snapshot vs actually deploy the app.  A quick Google seems to indicate others have experienced this error for various reasons so I'm not sure it's a DQD issue.  I'll also flag that we've been experiencing an issue with the latest version of one of DQD's dependencies, vroom (1.6.4).  So if you've got that version in your env I would recommend downgrading to 1.6.3, though I'm not sure that's the cause of your issue.

**Reply by sutsabs** [reply]
@katy-sadowski 
Thanks! I have posted in the Forum with the same question. If nothing what I can do is create a shiny app and use the DQD functions to load the results as for the DQ dashboard app.
